cli: dump command panics when table includes decimal with precision

```
$  grep -F '[config]' cockroach-data/logs/cockroach.INFO
I160714 10:33:43.355096 util/log/clog.go:998  [config] file created at: 2016/07/14 10:33:43
I160714 10:33:43.355096 util/log/clog.go:998  [config] running on machine: JESSEs-MacBook-Pro
I160714 10:33:43.355096 util/log/clog.go:998  [config] binary: CockroachDB beta-20160629-361-gf50771c (darwin amd64, built 2016/07/14 14:28:49, go1.6.2)
I160714 10:33:43.355096 util/log/clog.go:998  [config] arguments: [cockroach start]
```

What I did:

1. I started up a cluster.
2. I opened the interactive sql shell, created a table, and inserted some rows:
   ```
   root@:26257> create test.table dumptest (a int primary key, b decimal(10));
   CREATE TABLE

   root@:26257> insert into test.dumptest values (1, 1.1), (2, 2.2), (3, 3.3);
   INSERT 3
   ```
3. I tried to dump the contents of the new table:
   ```
   $ cockroach dump test dumptest
CREATE TABLE dumptest (
	a INT NOT NULL,
	b DECIMAL(10) NULL,
	CONSTRAINT "primary" PRIMARY KEY (a),
	FAMILY "primary" (a, b)
);
panic: unknown []byte type: 1.1, b: DECIMAL(10)

goroutine 1 [running]:
panic(0x5401380, 0xc82036c440)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
github.com/cockroachdb/cockroach/cli.dumpTable(0x749d330, 0xc820026010, 0xc82033f920, 0x7fff5fbffbc7, 0x4, 0x7fff5fbffbcc, 0x8, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/dump.go:205 +0x429b
github.com/cockroachdb/cockroach/cli.runDump(0x61b72c0, 0xc82033f8e0, 0x2, 0x2, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/dump.go:58 +0x16e
github.com/spf13/cobra.(*Command).execute(0x61b72c0, 0xc82033f720, 0x2, 0x2, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:600 +0x668
github.com/spf13/cobra.(*Command).ExecuteC(0x61b5720, 0x61b72c0, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:690 +0x55c
github.com/spf13/cobra.(*Command).Execute(0x61b5720, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/spf13/cobra/command.go:649 +0x2d
github.com/cockroachdb/cockroach/cli.Run(0xc82000a290, 0x3, 0x3, 0x0, 0x0)
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/cli/cli.go:95 +0x7c
main.main()
	/Users/jesseseldess/src/github.com/cockroachdb/cockroach/main.go:37 +0xe6
   ```

--- a/cli/dump.go
+++ b/cli/dump.go
@@ -192,15 +192,15 @@ func dumpTable(w io.Writer, conn *sqlConn, origDBName, origTableName string) err
 					switch ct := coltypes[cols[si]]; ct {
 					case "INTERVAL":
 						ivals[si] = fmt.Sprintf("'%s'", t)
-					case "DECIMAL":
-						ivals[si] = fmt.Sprintf("%s", t)
+					case "BYTES":
+						ivals[si] = parser.NewDBytes(parser.DBytes(t)).String()
 					default:
-						// STRING and BYTES types can have optional length
+						// STRING and DECIMAL types can have optional length
 						// suffixes, so only examine the prefix of the type.
 						if strings.HasPrefix(coltypes[cols[si]], "STRING") {
 							ivals[si] = parser.NewDString(string(t)).String()
-						} else if strings.HasPrefix(coltypes[cols[si]], "BYTES") {
-							ivals[si] = parser.NewDBytes(parser.DBytes(t)).String()
+						} else if strings.HasPrefix(coltypes[cols[si]], "DECIMAL") {
+							ivals[si] = string(t)
 						} else {
 							panic(errors.Errorf("unknown []byte type: %s, %v: %s", t, cols[si], coltypes[cols[si]]))
 						}

--- a/cli/dump_test.go
+++ b/cli/dump_test.go
@@ -55,7 +55,10 @@ func TestDumpRow(t *testing.T) {
 		n interval,
 		o bool,
 		e decimal,
-		tz timestamptz
+		tz timestamptz,
+		e1 decimal(2),
+		e2 decimal(2, 1),
+		s1 string(1)
 	);
 	INSERT INTO d.t VALUES (
 		1,
@@ -65,7 +68,10 @@ func TestDumpRow(t *testing.T) {
 		'2h30m30s',
 		true,
 		1.2345,
-		'2016-01-25 10:10:10'
+		'2016-01-25 10:10:10',
+		3.4,
+		4.5,
+		's'
 	);
 	INSERT INTO d.t VALUES (DEFAULT);
 	INSERT INTO d.t (f) VALUES (
@@ -96,18 +102,21 @@ CREATE TABLE t (
 	o BOOL NULL,
 	e DECIMAL NULL,
 	tz TIMESTAMP WITH TIME ZONE NULL,
-	FAMILY "primary" (i, f, d, t, n, o, tz, rowid),
+	e1 DECIMAL(2) NULL,
+	e2 DECIMAL(2,1) NULL,
+	s1 STRING(1) NULL,
+	FAMILY "primary" (i, f, d, t, n, o, tz, e1, e2, s1, rowid),
 	FAMILY fam_1_s (s),
 	FAMILY fam_2_b (b),
 	FAMILY fam_3_e (e)
 );
 
 INSERT INTO t VALUES
-	(1, 2.3, 'striiing', b'a1b2c3', '2016-03-26', '2016-01-25 10:10:10+00:00', '2h30m30s', true, 1.2345, '2016-01-25 10:10:10+00:00'),
-	(NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
-	(NULL, +Inf, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
-	(NULL, -Inf, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
-	(NULL, NaN, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+	(1, 2.3, 'striiing', b'a1b2c3', '2016-03-26', '2016-01-25 10:10:10+00:00', '2h30m30s', true, 1.2345, '2016-01-25 10:10:10+00:00', 3.4, 4.5, 's'),
+	(NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
+	(NULL, +Inf, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
+	(NULL, -Inf, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL),
+	(NULL, NaN, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
 `
 
 	if string(out) != expect {

